hashicorp/design-system
# Add Form Validation Docs

### :pushpin: Summary

If merged this PR will add further clarifying documentation to form patterns covering validation including:

- Types of validation (client/server-side): ways to validate forms from a technical perspective
- Methods of validation (delayed, on-submit): how validation is expressed to the user
- How to display validation (inline, form-level)

The additions in the documentation is an attempt to make the differentiation between how an application should approach validation from a technical perspective (via the type) and how the validation impacts the user (method, display).

### :link: External links


Jira ticket: [HDS-1609](https://hashicorp.atlassian.net/browse/HDS-1609)


[HDS-1609]: https://hashicorp.atlassian.net/browse/HDS-1609?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
